go-controller/pkg/ovn/master.go

```diff
@@ -1014,23 +1014,32 @@ func (oc *Controller) clearInitialNodeNetworkUnavailableCondition(origNode, newN
 }
 
 // delete chassis of the given nodeName/chassisName map
+// from chassis & chassis_private table
 func deleteChassis(ovnSBClient goovn.Client, chassisMap map[string]string) {
 	cmds := make([]*goovn.OvnCommand, 0, len(chassisMap))
 	for chassisHostname, chassisName := range chassisMap {
 		if chassisName != "" {
 			klog.Infof("Deleting stale chassis %s (%s)", chassisHostname, chassisName)
-			cmd, err := ovnSBClient.ChassisDel(chassisName)
+			chassisDelCmd, err := ovnSBClient.ChassisDel(chassisName)
 			if err != nil {
 				klog.Errorf("Unable to create the ChassisDel command for chassis: %s from the sbdb", chassisName)
 			} else {
-				cmds = append(cmds, cmd)
+				cmds = append(cmds, chassisDelCmd)
+			}
+			// also delete corresponding chassis row in chassis_private table
+			chassisPrivateDelCmd, err := ovnSBClient.ChassisPrivateDel(chassisName)
```

Comment:
is it worth checking to see if the running version of OVN has the private_chassis table before we try to delete to ensure backwards compatibility or no?

Reply:
I think it will be safe to do so.

I think this is an OVN bug in that when we delete a Chassis, I don't know why we leave behind Chassis_Private entry for that Chassis. Shouldn't it be deleted at the same time as Chassis deletion? @numansiddique @dceara for viz.

Reply:
Either 

1. we extend `go-ovn` to return the underlying TableSchema for OVN NB or OVN SB and then check that table to see if the connected OVN DB instance has the required table, and then proceed.
2. execute all the Chassis_Private commands separately and log a Warning on error as they are soft errors and will not cause any functional issue.

Reply:
@dceara from OVN K8s POV we would prefer not to delete the chassis_private record explicitly. So, it will be ideal if for Chassis_Private row to be removed automatically whenever its corresponding Chassis entry is deleted. This is because a Chassis_Private entry cannot exists by itself without a corresponding Chassis entry anyways. Or to delete from `ovn-northd` whenever it sees such inconsistencies.